packages/mui-base/src/SliderUnstyled/SliderUnstyled.js

```diff
@@ -246,53 +240,49 @@ const SliderUnstyled = React.forwardRef(function SliderUnstyled(props, ref) {
         const percent = valueToPercent(value, min, max);
         const style = axisProps[axis].offset(percent);
 
-        const ValueLabelComponent = valueLabelDisplay === 'off' ? Forward : ValueLabel;
-
         return (
-          <React.Fragment key={index}>
-            <ValueLabelComponent
-              {...(!isHostComponent(ValueLabelComponent) && {
-                valueLabelFormat,
-                valueLabelDisplay,
-                value:
-                  typeof valueLabelFormat === 'function'
-                    ? valueLabelFormat(scale(value), index)
-                    : valueLabelFormat,
-                index,
-                open: open === index || active === index || valueLabelDisplay === 'on',
-                disabled,
-              })}
-              {...valueLabelProps}
-              className={clsx(classes.valueLabel, valueLabelProps.className)}
-            >
-              <Thumb
-                data-index={index}
-                data-focusvisible={focusedThumbIndex === index}
-                {...thumbProps}
-                className={clsx(classes.thumb, thumbProps.className, {
-                  [classes.active]: active === index,
-                  [classes.focusVisible]: focusedThumbIndex === index,
+          <Thumb
+            key={index}
+            data-index={index}
+            data-focusvisible={focusedThumbIndex === index}
+            {...thumbProps}
+            className={clsx(classes.thumb, thumbProps.className, {
+              [classes.active]: active === index,
+              [classes.focusVisible]: focusedThumbIndex === index,
+            })}
+            style={{
+              ...style,
+              pointerEvents: disableSwap && active !== index ? 'none' : undefined,
+              ...thumbProps.style,
+            }}
+          >
+            <Input
+              data-index={index}
+              aria-label={getAriaLabel ? getAriaLabel(index) : ariaLabel}
+              aria-valuenow={scale(value)}
+              aria-labelledby={ariaLabelledby}
+              aria-valuetext={
+                getAriaValueText ? getAriaValueText(scale(value), index) : ariaValuetext
+              }
+              value={values[index]}
+              {...inputProps}
+            />
+            {ValueLabel ? (
```

Comment:
This would change the structure for the Material UI components. What was the reason for changing it? The ValueLabel should wrap the Thumb.

Reply:
> This would change the structure for the Material UI components. 

It won't change since SliderUnstyled is not used in Material UI's Slider component now. We use the `useSlider` hook and keep the JSX structure intact like the current SliderUnstyled in production. The `useSlider` hook is used to implement the `valueLabelDisplay` logic in Material UI Slider. 

> What was the reason for changing it?

If you inspect the generated HTML DOM structure, the value label is actually inside the thumb slot due to [this code](https://github.com/mui/material-ui/blob/master/packages/mui-base/src/SliderUnstyled/SliderValueLabelUnstyled.js#L27-L41) in the SliderValueLabelUnstyled. It means that even though the ValueLabel wraps the thumb in the JSX structure, in the final generated HTML the thumb wraps the value label due to `React.cloneElement`.  

Thus I changed the structure only in SliderUnstyled avoiding the complicated `React.cloneElement` implementation.

This is how Joy UI's Slider is also implemented. See https://github.com/mui/material-ui/blob/master/packages/mui-joy/src/Slider/Slider.tsx#L608-L639.

----

I would also recommend to read the PR description and also review `Slider.js` file in `packages/@mui-material` in this PR if you haven't.

Reply:
For easing up the review I would recommend splitting this in two pull requests. We can go first with replacing the `SliderUnstyled` with `useSlider` in @mui/material. It's important we don't have regressions with this change. After this one, we can safely do any changes we want to the `SliderUnstyled`, making sure it won't affect the Material UI's version. We aim to have each PR fix one particular problem, this would also make the review much easier.

> I would also recommend to read the PR description and also review Slider.js file in packages/@mui-material in this PR if you haven't.

I focused only on the changes inside `SliderUnstyled` in the initial review, haven't checked the other changes, I was only looking for potential breaking changes we may do.

Reply:
It makes sense. I will split it into two PRs.